src/Symfony/Component/Security/Core/Exception/CustomAuthenticationException.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Exception;
+
+/**
+ * An authentication exception where you can add a message that will
+ * be shown to the user, without needing to create a sub-class.
+ *
+ *  throw CustomAuthenticationException::createWithSafeMessage(
+ *      'That was a ridiculous username'
+ *  );
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+class CustomAuthenticationException extends AuthenticationException
+{
+    private $messageKey;
+
+    private $messageData;
+
+    /**
+     * Helper method to create this exception and set the safe message
+     * that will be shown to the user.
+     *
+     * @param string     $safeMessage
+     * @param array      $messageData
+     * @param int        $code
+     * @param \Exception $previous
+     *
+     * @return CustomAuthenticationException
+     */
+    static public function createWithSafeMessage($safeMessage = '', array $messageData = array(), $code = 0, \Exception $previous = null)
```

Comment:
We do not have such static methods to create exceptions in Symfony (unlike Doctrine).